foreshadow/steps/preprocessor.py

```diff
@@ -43,13 +49,32 @@ def fit(self, X, *args, **kwargs):
 
         list_of_tuples = []
 
+        # We need to handle the text data differently. Specifically, we will
+        # group them together and process them with in pipeline instead of
+        # one by one. In this pipeline, we want to apply a feature reducing
+        # component, for now, a PCA as an extra step. This is to prevent the
+        # TFIDF output to overwhelm the other columns. However, should we
+        # encapsulate the modification here or in the TextEncoder? I believe
+        # the latter makes more sense but we still need to separate the text
+        # features from the rest.
+
+        text_features = []
         for column in X.columns:
-            intent = self.cache_manager[AcceptedKey.INTENT, column]
-            transformation_pipeline = self._prepare_transformation_pipeline(
-                intent=intent, column=column
-            )
-            list_of_tuples.append((column, transformation_pipeline, column))
+            intent = self._get_intent(column=column)
+            if intent == Text.__name__:
+                text_features.append(column)
+            else:
+                transformation_pipeline = self.pipeline_by_intent[intent]
+                list_of_tuples.append(
+                    (column, transformation_pipeline, column)
+                )
 
+        text_transformation_pipeline = _configure_text_transformation_pipeline(
+            num_of_non_text_features=len(X.columns) - len(text_features)
+        )
+        list_of_tuples.append(
+            ("text_preprocessing", text_transformation_pipeline, text_features)
```

Comment:
Where does the merge of all text_features happen?

Reply:
Not here. There is a class called `DataSelector` where we merge text features. What happens here is that we send all text features to one transformation pipeline in one group.